lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -236,8 +238,21 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 			attrs["exitCode"] = code
 			log.Infof("Sending die event for container %s - code: %s", vc.ContainerID, code)
 			actor := CreateContainerEventActorWithAttributes(vc, attrs)
-
 			EventService().Log(containerDieEvent, eventtypes.ContainerEventType, actor)
+
+			// auto-remove if required
+			if vc.HostConfig.AutoRemove {
```

Comment:
HostConfig is available in the persona cache -- you are correct it doesn't persist across an appliance restart.  

If we execute `docker run -it --rm busybox` and then have an appliance reboot the container will come back up but the `autoremove` flag will be false.  If I reboot a docker host -- docker will remove the `autoremove` container, so that when the host is restarted the container will be gone.

How would you like to proceed?